pallets/pips/src/benchmarking.rs

```diff
@@ -0,0 +1,377 @@
+        let enact_call = Call::<T>::enact_snapshot_results(vec![(0, SnapshotResult::Approve)]);
+        enact_call.dispatch_bypass_filter(enact_origin)?;
+        let future_block = frame_system::Module::<T>::block_number() + 100.into();
+    }: _(proposer_origin, 0, Some(future_block))
+    verify {
+        assert_eq!(true, PipToSchedule::<T>::contains_key(&0));
+    }
+
+    clear_snapshot {
+        // length of the proposal padding
+        let c in 0 .. 100_000;
+
+        let (account, origin, did) = make_account::<T>("proposer", 0);
+        identity::CurrentDid::put(did);
+        let (proposal, url, description) = make_proposal::<T>(c as usize);
+        Module::<T>::set_proposal_cool_off_period(RawOrigin::Root.into(), 0.into())?;
+        Module::<T>::propose(
+            origin.clone().into(),
+            proposal,
+            42.into(),
+            Some(url.clone()),
+            Some(description.clone())
+        )?;
+        T::GovernanceCommittee::bench_set_release_coordinator(did);
+        Module::<T>::snapshot(origin.clone().into())?;
+        assert!(SnapshotMeta::<T>::get().is_some());
+    }: _(origin)
+    verify {
+        assert!(SnapshotMeta::<T>::get().is_none());
+    }
+
+    snapshot {
+        // length of the proposal padding
+        let c in 0 .. 100_000;
+
+        let (account, origin, did) = make_account::<T>("proposer", 0);
+        identity::CurrentDid::put(did);
+        let (proposal, url, description) = make_proposal::<T>(c as usize);
+        Module::<T>::set_proposal_cool_off_period(RawOrigin::Root.into(), 0.into())?;
+        Module::<T>::propose(
+            origin.clone().into(),
+            proposal,
+            42.into(),
+            Some(url.clone()),
+            Some(description.clone())
+        )?;
+        T::GovernanceCommittee::bench_set_release_coordinator(did);
+    }: _(origin)
+    verify {
+        assert!(SnapshotMeta::<T>::get().is_some());
+    }
+
+    enact_snapshot_results {
+        // length of the proposal padding
+        let c in 0 .. 100_000;
+
+        let (proposer_account, proposer_origin, proposer_did) = make_account::<T>("proposer", 0);
+        identity::CurrentDid::put(proposer_did);
+        let (proposal, url, description) = make_proposal::<T>(c as usize);
+        Module::<T>::set_proposal_cool_off_period(RawOrigin::Root.into(), 0.into())?;
+        Module::<T>::propose(
+            proposer_origin.clone().into(),
+            proposal,
+            42.into(),
+            Some(url.clone()),
+            Some(description.clone())
+        )?;
+        T::GovernanceCommittee::bench_set_release_coordinator(proposer_did);
+        Module::<T>::snapshot(proposer_origin.clone().into())?;
+        let enact_origin = T::VotingMajorityOrigin::successful_origin();
+        let enact_call = Call::<T>::enact_snapshot_results(vec![(0, SnapshotResult::Approve)]);
+    }: {
+        enact_call.dispatch_bypass_filter(enact_origin)?;
```

Comment:
same comment as above.

Reply:
Same answer as to your question above. It's only possible with weight adjustment in post-dispatch.

Reply:
post-dispatch is not a solution because that does not limit the unbounded complexity. It only refunds the users' fee when the function ends us being not so complex to execute. 

To solve the problem in this function, maybe we can make this a draining function. i.e. Every time this function is called, we process n records and remove them from the queue. This way, the complexity of one call will be limited to n records but we will still be able to process an unlimited number of records in total.

Alternatively, if we limit the size of a snapshot, this problem goes away. It's sensible to restrict snapshots to maybe 100 proposals at a time. Anything more is most probably going to be spam.

@Centril @adamdossa Thoughts?

Reply:
The benchmark should get at least these parameters as input:
1. the number of proposals in the snapshot queue (can be fixed to the maximum length of the queue for simplicity),
2. the number of proposals in `result` (already getting it),
3. the size of each proposal in `result` (can be fixed to the maximum size bound used by the benchmark).
4. whether each proposal in `result` is to be accepted or rejected.

I also simulated aye/nay votes in the benchmark setup but I believe this should be removed from input parameters because the numbers of votes are irrelevant.